modules/apps/forms-and-workflow/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/display/context/CalendarDisplayContext.java

```diff
@@ -41,13 +47,30 @@ public CalendarDisplayContext(
 		_themeDisplay = themeDisplay;
 	}
 
-	public List<Calendar> getOtherCalendars(long[] calendarIds)
+	public List<Calendar> getOtherCalendars(User user, long[] calendarIds)
 		throws PortalException {
 
 		List<Calendar> otherCalendars = new ArrayList<>();
 
 		for (long calendarId : calendarIds) {
-			Calendar calendar = _calendarService.fetchCalendar(calendarId);
+			Calendar calendar = null;
+			try {
+				calendar = _calendarService.fetchCalendar(calendarId);
+			}
+			catch (PrincipalException principalException) {
+				if (_log.isInfoEnabled()) {
+					StringBundler sb = new StringBundler(4);
```

Comment:
you can simple log Exception  _log.info(principalException);

Reply:
Yes,  _log.info(principalException) is simpler however, it only shows the permission error. It does not show why the error happened.  As Liferay best practices, we need to provide more context for the users, what is it, where it comes from, how to fix it for our end-user ( I am referring to developer users).